templates/waggle-dance-federation-local.yml.tmpl

```diff
@@ -4,5 +4,7 @@
     remote-meta-store-uris: thrift://${metastore_host}:${metastore_port}
 ${ mapped_databases == "" ? "" : "    mapped-databases:" }
 ${ mapped_databases == "" ? "" : join("\n",formatlist("    - %s",split(",",mapped_databases))) }
+${ database_name_mapping == "" ? "" : "    database-name-mapping:" }
+${ database_name_mapping == "" ? "" : join("\n", formatlist("      %s", split(",", replace(replace(database_name_mapping, " ", ""), ":", ": ")))) }
```

Comment:
Unrelated to this specific change but would it be better for longterm support of these templates to define the waggle-dance config as a `list(map)` in terraform & [yamlencode](https://www.terraform.io/docs/configuration/functions/yamlencode.html) it  or `jsonencode | yq .` before writing it to the file?

The formatting here seems painful to setup 😅

Reply:
Yes, I agree that this is painful.  When I was doing some changes for apiary-data-lake last year, we ran into this with passing tags into the module - a TF list of maps constrains the maps to only have string key/values.  You can switch to list(object) but then you can't have optional params.  The whole thing was a mess.

For that issue, went to using jsonencode when passing into the module, and then jsondecode in the module, which worked for that situation.

For this change, I did try something similar.  However, yamldecode (or jsondecode) is not allowed to be called within a resource of type `template_file` (it uses an old provider that doesn't support as much functionality).  You can use it in an inline `templatefile` but then I would have needed to restructure a lot of this module.  I didn't have the time to make such an invasive change for what I was hoping was a quick update to support database aliasing.

However, it is certainly something we might want to change in the future - we have many TF modules using `template_file` that should be ported to using `templatefile`.